modules/admin_manual/pages/enterprise/external_storage/enterprise_only_auth.adoc

```diff
@@ -1,45 +1,188 @@
 Global credentials::
-  Re-usable credentials entered by the admin, files can be shared.
+  Re-usable credentials entered by the admin, files can be shared. Global credentials are entered in a separate form.
++
+image:enterprise/external_storage/enterprise-only-auth/auth_backends-2.png[Global credentials form.]
 
-Global credentials are entered in a separate form.
+RSA public key::
+This authentication method is limited to SFTP only. Manually enter the public key or press btn:[Generate] in the mount form to generate one. Files can be shared.
 
-image:enterprise/external_storage/enterprise-only-auth/auth_backends-2.png[Global credentials form.]
+=== Authentication Options Allowing Sharing
 
-Use the dropdown selector to choose the authentication backend when you
-create a new external mount.
+The following table shows which authentication option allows sharing:
+ 
+[width="70%",cols="50%,^20%",",options="header"]
+|===
+| Authentication Option
+| Sharing Allowed
 
-image:enterprise/external_storage/enterprise-only-auth/auth_backends-1.png[Authentication dropdown selector.]
+| Username and password
+| x
+
+| Log-in credentials, save in session
+|
+
+| Log-in credentials, save in database
+| x
+
+| User entered, store in database
+| x
+
+| Credentials hardcoded in config file
+| x
+
+| Global credentials
+| x
+
+| RSA public key
+| x
+|===
+
+=== Notes for Credentials Hardcoded in Config File
+
+* The username taken will be the same as the `user id` of the ownCloud's session. With the example ownCloud's user "Alice" with password "mysecret" (or even without password) will use "Alice" as username and the password from the config.php file to access the storage. Note that for LDAP, the `user id` is usually like _1223-cbdf-cacc-1234...,_ unless the configuration in the user_ldap app is changed.
+
+* The password will be fetched from a key inside the config.php file, for details see the section below. Even if the account doesn't have a password like oAuth or OpenIDConnect ect, the connection with the storage will use the password from  config.php. *Note that the password will be the same for any user accessing the particular mount!*
+
+==== Defining key/value pairs in config.php
+
+Key/value pairs in config.php must have the following structure:
+
+.Example having a single array
+[source,php]
+----
+'customApp.config' => [
+    'password' => 'mypassword',
+]
+----
+
+.Example having nested arrays
+[source,php]
+----
+'customApp.config' => [
+  'server1' => [
+    'password' => 'mypassword',
+  ],
+  'server2' => [
+    ....
+  ],
+]
+----
+
+customApp.config::
+This can be any php conform name like the above `customApp.config` or as an another example `my.config.key`, but not reserved ownCloud keys.
```

Comment:
I am not sure what "conform" is referring to here. Maybe you intend to say that the "name" must be a string that is valid as an array key in a PHP array? If so, than actually any string could be used (even a UTF string in a different script). And looking down further, I see that using `#` in the key string will mess things up!)

Also, I don't understand why/when a structure of nested arrays would be needed/useful. Maybe there can be a realistic example given of a system that has multiple mounts, and how that could be structured in a clear way in the config.

Reply:
> the "name" must be a string that is valid as an array key in a PHP array

This is a much better wording

> looking down further, I see that using # in the key string will mess things up!

This was the definition made by @jvillafanez (see #4162 and wnd PR) to address a particular sub-key

> when a structure of nested arrays would be needed/useful

See the issue above. The use case is, when you have more than one server (sub-key) you define but collect it in one master key. I took the description from @jvillafanez

Reply:
I have improved the text and used your comments as input. Please have a look again.
If you are fine with, pls set this block to resolved.

Reply:
Looks good. I just reviewed a few other minor grammar/typos.